### PR TITLE
test: Update the new ZTEST APIs under tests/net/[m-p]*

### PR DESCRIPTION
**Pull request by hunterhu3000**
including the directory of
tests/net/mgmt
tests/net/mld
tests/net/neighbor
tests/net/net_pkt
tests/net/npf
tests/net/pm
tests/net/ppp
tests/net/promiscuous
tests/net/ptp

### CODE REVIEW

File: tests/net/mld/src/main.c [hunk 1]
```diff
@@ -167,7 +167,12 @@ static void group_joined(struct net_mgmt_event_callback *cb,
 static void group_left(struct net_mgmt_event_callback *cb,
 			 uint32_t nm_event, struct net_if *iface)
 {
-	if (nm_event != NET_EVENT_IPV6_MCAST_LEAVE) {
+	uint32_t expected_event = _NET_EVENT_IPV6_BASE |
```
Comment (rlubos) [inline tests/net/mld/src/main.c:170]: Isn't this changing the meaning of the test? Previously a specific event was expected but now more are allowed.

File: tests/net/mld/src/main.c [hunk 2]
```diff
@@ -547,21 +552,19 @@ static void test_solicit_node(void)
...
-			 ztest_unit_test(test_verify_join_group),
-			 ztest_unit_test(test_verify_leave_group),
-			 ztest_unit_test(test_catch_query),
-			 ztest_unit_test(test_verify_send_report),
-			 ztest_unit_test(test_allnodes),
-			 ztest_unit_test(test_solicit_node)
-			 );
-
-	ztest_run_test_suite(net_mld_test);
+	test_mld_setup();
```
Comment (rlubos) [inline tests/net/mld/src/main.c:557]: I think that putting all tests under a single test case might be confusing if something fails - you don't get now a clear information on which tests case failed in the report.

File: tests/net/mld/src/main.c (hunk 1 above)
Comment (hunterhu3000) [reply tests/net/mld/src/main.c]: You are right, this is a bug. I have corrected this.
But there is still some problem in the test case. The k_yield fucnction works not as expected. I have changed it to k_sleep.
Thanks to review.

File: tests/net/mld/src/main.c (hunk 2 above)
Comment (hunterhu3000) [reply tests/net/mld/src/main.c]: As some of the test cases are linked together, for example join group and leave group, I have combined them into groups and make them a test case.

File: tests/net/npf/src/main.c [hunk 3]
```diff
@@ -306,14 +306,13 @@ static void test_npf_eth_mac_addr_mask(void)
...
-			 ztest_unit_test(test_npf_example2),
-			 ztest_unit_test(test_npf_eth_mac_address),
-			 ztest_unit_test(test_npf_eth_mac_addr_mask));
-
-	ztest_run_test_suite(net_pkt_filter_test);
+	test_npf_iface();
+	test_npf_example1();
+	test_npf_example2();
+	test_npf_eth_mac_address();
+	test_npf_eth_mac_addr_mask();
```
Comment (rlubos) [inline tests/net/npf/src/main.c:315]: Could those tests be also executed as separate test cases?

File: tests/net/ppp/driver/src/main.c [hunk 4]
```diff
@@ -537,22 +537,20 @@ static void test_send_ppp_8(void)
...
+	test_send_ppp_pkt_with_full_and_partial();
+	test_ppp_verify_fcs_1();
+	test_ppp_calc_fcs_1();
+	test_ppp_verify_fcs_3();
+	test_send_ppp_3();
+	test_send_ppp_4();
+	test_send_ppp_5();
+	test_send_ppp_6();
+	test_send_ppp_7();
+	test_send_ppp_8();
```
Comment (rlubos) [inline tests/net/ppp/driver/src/main.c:553]: Ditto

File: tests/net/promiscuous/src/main.c [hunk 5]
```diff
@@ -380,16 +380,15 @@ static void test_verify_data(void)
...
-			 ztest_unit_test(test_set_promisc_mode_off_again));
-
-	ztest_run_test_suite(net_promisc_test);
+	test_iface_setup();
+	test_set_promisc_mode_on();
+	test_set_promisc_mode_on_again();
+	test_recv_data();
+	test_verify_data();
+	test_set_promisc_mode_off();
+	test_set_promisc_mode_off_again();
```
Comment (rlubos) [inline tests/net/promiscuous/src/main.c:391]: Ditto

File: tests/net/ptp/clock/src/main.c [hunk 6]
```diff
@@ -562,26 +562,28 @@ static void test_ptp_clock_get_user(void)
...
+{
+	test_check_interfaces();
+	test_address_setup();
+	test_ptp_clock_interfaces();
+	test_ptp_clock_iface_1();
+	test_ptp_clock_iface_2();
+	test_ptp_clock_get_by_index();
+	test_ptp_clock_get_by_index_user();
+	test_ptp_clock_get_kernel();
+	test_ptp_clock_get_user();
```
Comment (rlubos) [inline tests/net/ptp/clock/src/main.c:586]: Ditto

File: tests/net/npf/src/main.c (hunk 3 above)
Comment (hunterhu3000) [reply tests/net/npf/src/main.c:315]: only test_npf_example1 and test_npf_example2 can be split out.
I'll change it.

File: tests/net/ppp/driver/src/main.c (hunk 4 above)
Comment (hunterhu3000) [reply tests/net/ppp/driver/src/main.c:553]: when running separately, test_send_ppp_4 got problem. Will debug on it.

File: tests/net/ptp/clock/src/main.c (hunk 6 above)
Comment (hunterhu3000) [reply tests/net/ptp/clock/src/main.c:586]: This test case is not able to be run separately. The next function is based on the result of the previous function. They share the same global variables.

File: tests/net/ppp/driver/src/main.c (hunk 4 above)
Comment (hunterhu3000) [reply tests/net/ppp/driver/src/main.c:553]: These test cases use the same ppp interface, but they are not thread safe. So they can only be run one by one, not able to separate.

File: tests/net/promiscuous/src/main.c (hunk 5 above)
Comment (hunterhu3000) [reply tests/net/promiscuous/src/main.c:391]: This test case is not able to be run separately. The next function is based on the result of the previous function. They share the same global variables.

File: tests/net/ppp/driver/src/main.c (hunk 4 above)
Comment (rlubos) [reply tests/net/ppp/driver/src/main.c:553]: Ok, understood.

**Review by rlubos — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Pull request merged**